EBGodunov/hydro_ebgodunov_edge_state_2D.cpp

```diff
@@ -186,23 +186,33 @@ EBGodunov::ComputeEdgeState ( Box const& bx, int ncomp,
             // If we can't compute good transverse terms, don't use any d/dt terms at all
             if (apy(i-1,j,k) > 0. && apy(i-1,j+1,k) > 0.)
             {
-                Real quxl = (apx(i,j,k)*u_mac(i,j,k) - apx(i-1,j,k)*u_mac(i-1,j,k)) * q(i-1,j,k,n);
-                stl += ( - (0.5*dtdx) * quxl
-                         - (0.5*dtdy) * (apy(i-1,j+1,k)*yzlo(i-1,j+1,k  ,n)*v_mac(i-1,j+1,k  )
-                                        -apy(i-1,j  ,k)*yzlo(i-1,j  ,k  ,n)*v_mac(i-1,j  ,k  )) ) / vfrac_arr(i-1,j,k);
-                if (fq && vfrac_arr(i-1,j,k) > 0.)
-                    stl += 0.5*l_dt*fq(i-1,j,k,n);
+                if (iconserv[n]) {
+                    Real quxl = (apx(i,j,k)*u_mac(i,j,k) - apx(i-1,j,k)*u_mac(i-1,j,k)) * q(i-1,j,k,n);
+                    stl += ( - (0.5*dtdx) * quxl
+                             - (0.5*dtdy) * (apy(i-1,j+1,k)*yzlo(i-1,j+1,k  ,n)*v_mac(i-1,j+1,k  )
+                                            -apy(i-1,j  ,k)*yzlo(i-1,j  ,k  ,n)*v_mac(i-1,j  ,k  )) ) / vfrac_arr(i-1,j,k);
+                    if (fq && vfrac_arr(i-1,j,k) > 0.)
+                        stl += 0.5*l_dt*fq(i-1,j,k,n);
```

Comment:
@esclapez What of the external forcing piece (`+=0.5*l_dt*fq(i-1,j,k,n)`) for non-conservative? Can we really just drop it?

Reply:
I don't think we can just drop it.

Reply:
Good catch, oversight on my side. The if (fq ...) should be moved outside of the test on iconserv. I tested the code with the forcing outside and went back to clean the code afterward.